contracts/base/LendingPool.sol

```diff
@@ -0,0 +1,217 @@
+/*
+    Copyright 2020 Fabrx Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+pragma solidity 0.5.17;
+
+// Libraries
+import "../util/AddressLib.sol";
+
+// Commons
+import "../util/ZeroCollateralCommon.sol";
+
+// Interfaces
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "../interfaces/LendingPoolInterface.sol";
+import "../interfaces/LendersInterface.sol";
+import "../interfaces/ZTokenInterface.sol";
+
+// Contracts
+import "./Initializable.sol";
+
+
+/**
+    @notice The LendingPool contract holds all of the tokens that lenders transfer into the protocol. It is the contract that lenders interact with to deposit and withdraw their tokens including interest. The LendingPool interacts with the LenderInformation contract to ensure token balances and interest owed is kept up to date.
+ */
+contract LendingPool is LendingPoolInterface, Initializable {
+    using AddressLib for address;
+
+    /* State Variables */
+
+    IERC20 public token;
+
+    ZTokenInterface public zToken;
+
+    LendersInterface public lenders;
+
+    address public loans;
+
+    /** Modifiers */
+
+    /**
+        @notice It checks the address is the Loans contract address.
+        @dev It throws a require error if parameter is not equal to loans contract address.
+     */
+    modifier isLoan() {
+        loans.requireEqualTo(msg.sender, "Address is not Loans contract.");
+        _;
+    }
+
+    /* Constructor */
+
+    /** External Functions */
+
+    /**
+        @notice It initializes the contract state variables.
+        @param zTokenAddress zToken token address.
+        @param tokenAddress ERC20 token address.
+        @param lendersAddress Lenders contract address.
+        @param loansAddress Loans contract address.
+        @dev It throws a require error if the contract is already initialized.
+     */
+    function initialize(
+        address zTokenAddress,
+        address tokenAddress,
```

Comment:
I think maybe to differentiate between `zToken` and `token` more clearly, they should be called `zToken` and `lendingToken` ?